cmd/dns_list.go

```diff
@@ -57,37 +51,33 @@ Supported output template annotations: %s`,
 }
 
 func listDomains(filters []string) (outputter, error) {
-	domains, err := csDNS.GetDomains(gContext)
+	domains, err := cs.ListDNSDomains(gContext, gCurrentAccount.DefaultZone)
 	if err != nil {
 		return nil, err
 	}
 
 	out := dnsListOutput{}
 
 	for _, d := range domains {
-		keep := true
-		if len(filters) > 0 {
-			keep = false
-			s := strings.ToLower(fmt.Sprintf("%d#%s#%s", d.ID, d.Name, d.UnicodeName))
-
-			for _, filter := range filters {
-				substr := strings.ToLower(filter)
-				if strings.Contains(s, substr) {
-					keep = true
-					break
-				}
-			}
+		o := dnsListItemOutput{
+			ID:   StrPtrFormatOutput(d.ID),
+			Name: StrPtrFormatOutput(d.UnicodeName),
 		}
 
-		if !keep {
-			continue
+		if len(filters) == 0 {
+			out = append(out, o)
+			break
 		}
 
-		out = append(out, dnsListItemOutput{
-			ID:          d.ID,
-			Name:        d.Name,
-			UnicodeName: d.UnicodeName,
-		})
+		s := strings.ToLower(fmt.Sprintf("%s#%s", o.ID, o.Name))
+
+		for _, filter := range filters {
+			substr := strings.ToLower(filter)
+			if strings.Contains(s, substr) {
+				out = append(out, o)
+				break
```

Comment:
```suggestion
				continue
```

Reply:
I think it should be `break`, or at least that is how i am interpreting filters feature. This runs in nested loop, once first filter (substring) matches we will add it to output and move to the next domain. If `continue` it will try to match next filter on already added domain and may add it twice.

Reply:
ah yes, right!